### DISCUSSION TITLE
Search Box doesn't filter when result Source Id limited to "Documents"

### DISCUSSION THREAD

**Discussion by Zuckabub** [microsoft-search/pnp-modern-search#2074] [Q&A] [answered]
I have a page that has Search Results web part configured to use input query text from Search Box on the same page.

When I don't set "Result Source Id / Scope|Name" in the Search Results webpart, the search box works as I'd expect it to. 
When I set the value of "Result Source Id / Scope|Name" to "Documents", even though the results are limited to documents, the search box doesn't seem to filter the results anymore - no matter what I enter in the search box I get all results back.

Is this a bug or have I not configured things correctly? Essentially I want users to be able to search all documents they have access to.

**Comment by wobba** [comment, marked as answer]
That's because the "Documents" result source don't have {searchTerms} as a query variable. The result source "Documents" probably won't do what you expect and I suggest you craft your own KQL to return what you want to have returned.

This is the query template for Documents `{?path:{Scope}} {?owstaxIdMetadataAllTagsInfo:{Tag}} (FileExtension:doc OR FileExtension:docx OR FileExtension:xls OR FileExtension:xlsx OR FileExtension:ppt OR FileExtension:pptx OR FileExtension:pdf) (IsDocument:"True" OR contentclass:"STS_ListItem")`

**Reply by Zuckabub** [reply]
Thank you - Using my own KQL query as suggested allowed me to get the results I wanted. 
This is the content of my query template configuration:
`{searchTerms} isDocument:1`
"Result Source Id / Scope|Name" is empty.

**Reply by wobba** [reply]
Fyi, IsDocument will exclude some file formats, but you can adjust the KQL filter as needed.
